### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'getObjectType')

### DISCUSSION THREAD

**Discussion by FarhanRafid97** [MichalLytek/type-graphql#1299] [Q&A]
why when i add @FieldResolver my app crash 

@FieldResolver(() => String, { defaultValue: {} })
  email(@Root() user: User, @Ctx() { req }: MyContext) {
    if (req.session.userId === user.id) {
      return user.email;
    }
    return '';
  }

**Comment by MichalLytek** [comment]
Please create a minimal reproducible code example repository.

**Comment by antontsvil** [comment]
Ran into this same issue. It was due to the `@FieldResolver` being in the wrong place. It should be placed under the _resolver_ (decorated with `@Resolver`), and not the _type_ (decorated with `@ObjectType`)
